### PR TITLE
feat: Convert fns module to TS

### PR DESCRIPTION
**Pull request by yavorona**
## Summary

- Convert `fns` module from JS to TS.

## Test plan
- Existing unit tests + running with local bundle

### CODE REVIEW

File: packages/optimizely-sdk/lib/utils/fns/index.ts [hunk 1]
```diff
@@ -15,20 +15,23 @@
  */
 import { generateUUID as uuid, keyBy as keyByUtil } from '@optimizely/js-sdk-utils';
 
-var MAX_SAFE_INTEGER_LIMIT = Math.pow(2, 53);
+import { Config } from '@optimizely/optimizely-sdk';
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/utils/fns/index.ts:18]: I believe this should be the right way of importing `Config` interface as it compiles successfully with `npm run build`. However, my TS compiler is failing with `Cannot find module '@optimizely/optimizely-sdk'` error, not sure why. Any ideas @mjc1283? 🥺

Comment (yavorona) [reply packages/optimizely-sdk/lib/utils/fns/index.ts]: This is now outdated. I changed this to type `any` after discussing with @mikeng13 . Please see the most recent commit for reference.

File: packages/optimizely-sdk/lib/utils/fns/index.ts [hunk 2]
```diff
@@ -15,20 +15,21 @@
  */
 import { generateUUID as uuid, keyBy as keyByUtil } from '@optimizely/js-sdk-utils';
 
-var MAX_SAFE_INTEGER_LIMIT = Math.pow(2, 53);
+const MAX_SAFE_INTEGER_LIMIT = Math.pow(2, 53);
 
-export var assign = function (target) {
-  if (!target) {
+// eslint-disable-next-line
+export function assign(...args: [any]): any {
```
Comment (mjc1283) [inline packages/optimizely-sdk/lib/utils/fns/index.ts:21]: Is there a better type signature we can use here? The type of `Object.assign` is `assign<T, U>(target: T, source: U): T & U`.

File: packages/optimizely-sdk/lib/utils/fns/index.ts [hunk 3]
```diff
@@ -15,20 +15,21 @@
  */
 import { generateUUID as uuid, keyBy as keyByUtil } from '@optimizely/js-sdk-utils';
 
-var MAX_SAFE_INTEGER_LIMIT = Math.pow(2, 53);
+const MAX_SAFE_INTEGER_LIMIT = Math.pow(2, 53);
 
-export var assign = function (target) {
+// eslint-disable-next-line
+export function assign(target: any, ...sources: any[]): any {
```
Comment (yavorona) [inline packages/optimizely-sdk/lib/utils/fns/index.ts:21]: @mjc1283 let me know if that works. I assumed we still want to go with `any` is this case.

**Pull request merged**